chatkit-core/src/test/kotlin/com/pusher/chatkit/rooms/state/LeftRoomTest.kt

```diff
@@ -0,0 +1,37 @@
+package com.pusher.chatkit.rooms.state
+
+import assertk.assertThat
+import assertk.assertions.isNotNull
+import com.pusher.chatkit.state.ChatkitState
+import com.pusher.chatkit.state.LeftRoom
+import org.spekframework.spek2.Spek
+import org.spekframework.spek2.style.specification.describe
+
+class LeftRoomTest : Spek({
+
+    describe("given a joined rooms state of two rooms") {
+        val givenState = ChatkitState(
+                joinedRoomsState = JoinedRoomsState(
+                        mapOf(
+                                JoinedRoomsStateTestUtil.roomOneId to JoinedRoomsStateTestUtil.roomOne,
+                                JoinedRoomsStateTestUtil.roomTwoId to JoinedRoomsStateTestUtil.roomTwo
+                        ),
+                        mapOf(
+                                JoinedRoomsStateTestUtil.roomOneId to 1,
+                                JoinedRoomsStateTestUtil.roomTwoId to 2
+                        )
+                )
+        )
+
+        describe("when an event for leaving a room that is part of the state is received") {
+            val newState = leftRoomReducer(givenState,
+                    LeftRoom(JoinedRoomsStateTestUtil.roomOneId))
+
+            it("then the state should be empty") {
```

Comment:
the then part is no longer correct, I would keep the whole BDD description terse and clear as:
```
given two rooms
  when left room
    then the state contains the expected room
```

Reply:
👍 good spot - have fixed

Reply:
cool, thanks, how about making given and then parts more concise? I forgot to explicitly mention it but also lets drop "should" from then parts to make them more concise and to clearly describe the behaviour (not a desire/wish/expectation -- Kevlin Henney had a talk where he covered that)